From a879b350b9669989277b570f6c2af92a4265c0ef Mon Sep 17 00:00:00 2001
From: AnNDP <AnNDP@fsoft.com.vn>
Date: Wed, 6 Jun 2018 03:15:41 -0400
Subject: [PATCH 6/9] waylandsink: Don't wait for the wl_surface_frame callback
 to notify the compositor of a wl_buffer

When the frame rendering delays, the weston compositor can drop frames
and catch up with the on-time processing by itself.

Signed-off-by: AnNDP <AnNDP@fsoft.com.vn>
---
 ext/wayland/gstwaylandsink.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 111cdab..6a25295 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -726,12 +726,6 @@ gst_wayland_sink_show_frame (GstVideoSink * vsink, GstBuffer * buffer)
     }
   }
 
-  /* drop buffers until we get a frame callback */
-  if (sink->redraw_pending) {
-    GST_LOG_OBJECT (sink, "buffer %p dropped (redraw pending)", buffer);
-    goto done;
-  }
-
   /* make sure that the application has called set_render_rectangle() */
   if (G_UNLIKELY (sink->window->render_rectangle.w == 0))
     goto no_window_size;
-- 
1.9.1

